### ISSUE TITLE
(dyoon): support non-4D tensors in the c ost functions of convolution 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#2824]
related ops (Conv, Pool, BatchNorm, and their backprops) and the related
helper functions.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [480ac84aa8390e19a54bd2feef3a6069d959bb4e](https://github.com/imdone/tensorflow/commit/480ac84aa8390e19a54bd2feef3a6069d959bb4e) by A. Unique TensorFlower gardener@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/core/grappler/costs/op_level_cost_estimator.cc#L90-L97
